### ISSUE TITLE
Technical specification for zero-knowledge proof token transfer on Plasma

### ISSUE THREAD

**Issue by vrde** [leapdao/meta#204]
# Bounty
TL;DR: privacy for the masses!

Leap network is an implementation of Plasma that provides a cheap, fast, secure, and efficient way to transfer ERC20 tokens and execute spending conditions. Given those properties, leap network is the perfect soil to build payment networks and run expensive computations.

Projects like [ZoKrates](https://zokrates.github.io/) allow to model generic zero-knowledge proofs on Ethereum by providing an easy way to generate a solidity smart contract. This contract is stateless, its role is to only evaluate statements and return true or false. The contract can be then extended to actually do something with the result of the computation. For the scope of this bounty, ZoKrates (et simila) can be used to improve user's privacy by allowing token transfers without revealing the details about those transfers.

Computing a zero-knowledge proof is expensive in terms of gas, that's why running it on Leap network would make a lot of sense.

For the future, integrating zero-knowledge transfers in a tool like the burner wallet would give super powers to users of the Leap network by allowing them to have a **usable** way to make payments while **preserving their privacy**. :boom: 

This bounty is to analyze the feasibility of running a zero-knowledge payment network on plasma, and write a technical specification on how to implement that.

## Scope
- Define to which extent privacy is preserved. Given that tokens has to be transferred to the bridge contract, privacy can only be ensured from that point on. Users can always use mixers in the main Ethereum network.
- Analyze the status quo of zero-knowledge frameworks to see which one would fit this use case better.
- Check the status of other projects on the same topic (a simple online research shows that there is work on [ERC1723](https://github.com/ethereum/EIPs/issues/1723)).

## Deliverables
A technical specification that describes the details and steps to implement a zero-knowledge token transfer on plasma.

## Gain for the project
LeapDAO, by championing this initiative, would be seen as a fast, privacy preserving payment network. This would increase visibility of the project and drive adoption of the Leap network.

## Roles
bounty gardener: @vrde / 10%
bounty worker: name / share
bounty reviewer: name / share

**Comment by johannbarbie** [comment]
timeout
